src/entity_state.rs

```diff
@@ -0,0 +1,33 @@
+use bevy::prelude::Component;
+
+#[derive(Default)]
+pub enum EntityDirection {
```

Comment:
Pedantic suggestion: these things probably shouldn't be Entity\<whatever\>, since not all entities might have a direction, movement, ground state, etc. The level e.g. is an entity, so probably could rename.